pkg/p2p/peer/peer.go

```diff
@@ -316,17 +273,19 @@ func (p *Reader) readLoop() {
 			return
 		}
 
-		// TODO: error here should be checked in order to decrease reputation
-		// or blacklist spammers
-		startTime := time.Now().UnixNano()
-		if err := p.router.Collect(message); err != nil {
-			l.WithError(err).Error("message routing")
-		}
+		go func() {
```

Comment:
Spawning a goroutine to handle a request is pretty idiomatic approach. But again if the idea of this PR is to reduce concurrency not sure how this is achieved if you're introducing hundreds of new goroutines (consider 9 peer connections processing at least 10 wire messages at a time )

Reply:
In addition, not sure we need to gain more technical debt in P2P layer as we tend to migrate to Kadcast.

Reply:
To be more clear - the goal is to reduce concurreny where it is not necessary, so as to avoid potential race conditions, and simplify code. If concurrency can be used to its advantage, there is definitely room for it, which is why I use it here. For this same reason, I did not get rid of inter-process communication performed between, for instance, the mempool and the blockgenerator. Neither did I get rid of the channels used in the consensus for event processing, since the pattern actually works really well there and I don't think it would benefit from a different design. You've said it yourself, spawning goroutines to handle requests is an idiomatic approach, and it was preferred this way when it was mentioned on the stand-up a few days ago, to avoid bottlenecking. It does increase the number of goroutines, but the processing pipeline for many of the messages have become synchronous in isolation, the aim of which, again, was to avoid potential data races.